plugin/vcs/github/github.go

```diff
@@ -0,0 +1,286 @@
+	return errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) ReadFile(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath, commitID string) (string, error) {
+	return "", errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) ReadFileMeta(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath, branch string) (*vcs.FileMeta, error) {
+	return nil, errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) CreateWebhook(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID string, payload []byte) (string, error) {
+	return "", errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) PatchWebhook(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, webhookID string, payload []byte) error {
+	return errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) DeleteWebhook(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, webhookID string) error {
+	return errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func httpGet(ctx context.Context, client *http.Client, resourcePath string, token *string, oauthCtx oauthContext, refresher common.TokenRefresher) (code int, respBody string, err error) {
+	return retry(ctx, client, token, oauthCtx, refresher,
+		func() (*http.Response, error) {
+			url := fmt.Sprintf("%s/%s", apiURL, resourcePath)
+			req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+			if err != nil {
+				return nil, errors.Wrapf(err, "construct GET %s", url)
+			}
+
+			req.Header.Set("Content-Type", "application/json")
+			req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", *token))
+			resp, err := client.Do(req)
+			if err != nil {
+				return nil, errors.Wrapf(err, "GET %s", url)
+			}
+			return resp, nil
+		},
+	)
+}
+
+const maxRetries = 3
+
+func retry(ctx context.Context, client *http.Client, token *string, oauthCtx oauthContext, refresher common.TokenRefresher, f func() (*http.Response, error)) (code int, respBody string, err error) {
+	var resp *http.Response
+	var body []byte
+	for retries := 0; retries < maxRetries; retries++ {
+		select {
+		case <-ctx.Done():
+			return 0, "", ctx.Err()
+		default:
+		}
+
+		resp, err = f()
+		if err != nil {
+			return 0, "", err
+		}
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return 0, "", errors.Wrapf(err, "read response body with status code %d", resp.StatusCode)
+		}
+
+		if err = getOAuthErrorDetails(resp.StatusCode, string(body)); err != nil {
+			if _, ok := err.(*oauthError); ok {
```

Comment:
GitLab error checking is not working from what I understand, because `getOAuthErrorDetails` always returns a pointer of `oauthError`:

https://github.com/bytebase/bytebase/blob/4294b21a1f9c490791dbf7a8b4e1a929b419ff57/plugin/vcs/gitlab/gitlab.go#L700

Reply:
Yeah, we need another retryer.